scheduler/src/cook/scheduler/scheduler.clj

```diff
@@ -1084,19 +1084,64 @@
                               :first-ten-jobs (print-str (->> jobs (take 10) (map :job/uuid) (map str)))}))
       (dissoc compute-cluster->jobs :no-acceptable-compute-cluster))))
 
+(defn decrement-scheduling-capacity-counter
+  "Decrement capacity for compute cluster in the counter."
+  [capacity-counter compute-cluster]
+  (update-in capacity-counter [compute-cluster] dec))
+
+(defn scheduling-capacity-constrained-job-distributor
+  "Create a job distributor function that only assigns jobs to compute clusters
+   with available scheduling capacity. Assignments are tracked and the given
+   counter is decremented."
+  [compute-cluster->scheduling-capacity autoscalable-jobs pool-name compute-clusters job->acceptable-compute-clusters-fn]
+  (tracing/with-span [s {:name "scheduler.distribute-jobs-to-compute-cluster"
+                         :tags {:pool pool-name :component tracing-component-tag}}]
+    (let [capacity-counter-atom (atom compute-cluster->scheduling-capacity)
+          compute-cluster->jobs
+          (group-by
+           (fn choose-compute-cluster-for-autoscaling
+             [{:keys [job/uuid] :as job}]
+             (let [acceptable-compute-clusters
+                   (job->acceptable-compute-clusters-fn job compute-clusters)
+                   available-compute-clusters
+                   (filter
+                    (fn [compute-cluster]
+                      (let [capacity (get @capacity-counter-atom compute-cluster)]
+                        (when (nil? capacity)
+                          (throw (ex-info "Compute cluster given to distributor is not tracking scheduling capacity"
+                                          {:pool-name pool-name :compute-cluster compute-cluster})))
+                        (pos? capacity)))
+                    acceptable-compute-clusters)]
+               (if (empty? available-compute-clusters)
+                 :no-acceptable-compute-cluster
```

Comment:
I was mulling over if we should pause here. If we have a job that does have CC it can go, but that place doesn't have any available capacity, do we skip and try again later, or pause. Either way we can't place the job, and if there are too many we'll clog the queue.  But that's expected. Better to let what can match match and hope for luck soon.